src/app/pwa/_server/persistence/pools/blockchain/get-contract-user-pools.ts

```diff
@@ -69,14 +69,18 @@ export async function getUserPools(userAddress: Address) {
         })),
     })
 
+    // Get the offset in minutes for the user's timezone
+    const now = new Date();
+    const offsetInMinutes = now.getTimezoneOffset();
+
     return poolDetailsResults
         .map((result, index) => {
             if (result.status === 'success') {
                 const [, poolDetail, , poolStatus, , participants] = result.result
                 return {
                     id: allUserPools[index].toString(),
-                    timeStart: Number(poolDetail.timeStart),
-                    timeEnd: Number(poolDetail.timeEnd),
+                    timeStart: Number(poolDetail.timeStart) - offsetInMinutes * 60,
```

Comment:
We are handling the offsets in the server, I am not sure if the server timezone will match with the user one. Did you properly test it?

Reply:
Oh shoot, I just tested the value didn't notice it's server, let me change this at client side instead.